src/v/cluster_link/replication/tests/deps_test_impl.cc

```diff
@@ -49,7 +49,11 @@ raft::replicate_stages accounting_sink::replicate(
 
 void accounting_sink::notify_replicator_failure(model::term_id) {}
 
-ss::future<> random_data_source::start() noexcept { return ss::now(); }
+ss::future<> random_data_source::start(kafka::offset offset) noexcept {
+    _next = offset;
```

Comment:
The current pattern of 

```
start()
reset(offset) (aka assign(offset))
fetch()
```

is a totally valid scenario right? The fetch should start from `offset`. I'm not opposed to making this change but I have a feeling this is masking an underlying consumer issue where assign() semantics are not working as intended.